plotly/tools.py

```diff
@@ -1443,30 +1443,320 @@ class FigureFactory(object):
     """
 
     @staticmethod
-    def create_scatterplotmatrix(df, useindex = False, index = " ",
-                                 diagonal = "Scatter", symbol = 0, size = 6,
-                                 height = " ", width = " ", jitter = 0,
-                                 title = "Scatterplot Matrix"):
+    def _scatterplot_no_index(matrix, headers, 
+                              diag, size, symbol,
+                              height, width,
+                              jitter, title):
+        from plotly.graph_objs import graph_objs
+        from random import gauss
+        
+        dim = len(matrix)
+        fig = make_subplots(rows=dim, cols=dim)
+        trace_list = []
 
-        from plotly import tools
-        import plotly.plotly as py
-        import plotly.graph_objs as go
-        import pandas as pd
-        import random as r
-       
-        colors = ["rgb(31, 119, 180)", "rgb(255, 127, 14)",
-                  "rgb(44, 160, 44)", "rgb(214, 39, 40)",
-                  "rgb(148, 103, 189)", "rgb(140, 86, 75)",
-                  "rgb(227, 119, 194)", "rgb(127, 127, 127)",
-                  "rgb(188, 189, 34)", "rgb(23, 190, 207)"]
+        # Insert traces into trace_list
+        for listy in matrix:
+            for listx in matrix:
+                if (listx == listy) and (diag == "Histogram"):
+                    trace = graph_objs.Histogram(
+                        x = listx,
+                        showlegend = False
+                        )
+                elif (listx == listy) and (diag == "box"):
+                    trace = graph_objs.Box(
+                        y = listx,
+                        name = None,
+                        showlegend = False
+                        )
+                else:
+                    # Add Jitter
+                    if not isinstance(listx[0], basestring):
+                        spanx = abs(max(listx) - min(listx))
+                        for j in range(len(listx)):
+                            listx[j] = listx[j] + (spanx/2)*jitter*gauss(0,1)
+                    if not isinstance(listy[0], basestring):
+                        spany = abs(max(listy) - min(listy))
```

Comment:
Do you need to check both `listy` and `listx` here? It's possible that the division below will fail, not sure if that's what you want.

I'd either (1) check that they're *both* what you want or (2) wrap this is a try-except block. It's unclear why we know to do this operation is something is *not* a string.